src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -292,7 +292,15 @@ protected virtual void FinalizeRuntimeSpecificStaticFieldLayout(TypeSystemContex
         {
         }
 
-        private static ComputedInstanceFieldLayout ComputeExplicitFieldLayout(MetadataType type, int numInstanceFields)
+        /// <summary>
+        /// Called during instance field layout to setup initial contents of reference types. CoreCLR puts
+        /// an implicit pointer at the beginning of each reference type object.
+        /// </summary>
+        protected virtual void PrepareRuntimeSpecificSystemObjectLayout(TypeSystemContext context, ref LayoutInt instanceBytesBeforeFirstField, ref LayoutInt alignmentRequirement)
```

Comment:
This extension point allows someone to put Sequential/Explicit layout on `System.Object` and add actual fields to it that we will then lay out using the usual algorithm. Is this a scenario we need to support?

I would just add an algorithm class for System.Object, like we have for [__UniversalCanon](https://github.com/dotnet/corert/blob/c8a9707235d02e7cfc534bf2042cd4da3911b3b4/src/Common/src/TypeSystem/Common/UniversalCanonLayoutAlgorithm.cs), check for `System.Object` in ready to run's type system context `GetLayoutAlgorithmForType` override, and return that algorithm if it's `object`.

Reply:
Love it